src/Swarm/TUI/Controller.hs

```diff
@@ -516,3 +522,29 @@ handleInfoPanelEvent s (VtyEvent ev) = do
       let s' = s & uiState . uiInventory . _Just . _2 .~ l'
       continue s'
 handleInfoPanelEvent s _ = continueWithoutRedraw s
+
+-- | Handle user input events searching in the info panel.
+handleSearchingEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
+handleSearchingEvent s (VtyEvent (V.EvKey V.KEsc []))
+  = continue $ s & uiState . uiSearching .~ False
+                 & uiState . uiSearch .~ ""
+                 & uiState . uiInventory . _Just . _1 .~ 0
+
+handleSearchingEvent s (VtyEvent (V.EvKey V.KEnter []))
+  = continue $ s & uiState . uiSearching .~ False
+                 & uiState . uiInventory . _Just . _1 .~ 0
+
+handleSearchingEvent s (VtyEvent (V.EvKey (V.KChar c) []))
+  = continue $ s & uiState . uiSearch %~ (`T.snoc` c)
+                 & uiState . uiInventory . _Just . _1 .~ 0
+
+handleSearchingEvent s (VtyEvent (V.EvKey V.KBS []))
+  = continue $ s & uiState . uiSearch %~ safeTextInit
+                 & uiState . uiInventory . _Just . _1 .~ 0
+
+handleSearchingEvent s _ = continueWithoutRedraw s
```

Comment:
I think we should handle navigation event (e.g. up/down) so that we can select the desired item while searching.

Reply:
Oh, I think I can pass those events on to the pane and it might do the right thing.